br/pkg/lightning/mydump/loader.go

```diff
@@ -99,6 +102,76 @@ func (m *MDTableMeta) GetSchema(ctx context.Context, store storage.ExternalStora
 	return string(schema), nil
 }
 
+// MDLoaderSetupConfig stores the configs when setting up a MDLoader.
```

Comment:
The setup procedure seems:
1. construct a mdLoaderSetupOptsOption with MDLoaderSetupOption (implemented by maxScanFilesMDLoaderSetupOption)
2. create a MDLoaderConfig (which has a SetupOptions array) and apply the mdLoaderSetupOptsOption to it. The maxScanFilesMDLoaderSetupOption is conveyed to the SetupOptions array.
3. create a SetupConfig with field MaxScanFiles and apply the SetupOptions from MDLoaderConfig.
4. The SetupConfig.MaxScanFiles = maxScanFilesMDLoaderSetupOption.maxScanFiles

The setup procedure is a bit too complicated with a bunch of seemingly redundant abstractions, which only ONE class implements for each. 

I understand the general interface is conducive for future expansion, but currently, the MDLoaderConfig only serves for the SetupConfig, so it might work as well by directly passing the MDLoaderSetupOption from outside instead of the MDLoaderOption so that the code would look more succinct.

Alternatively, you are supposed to leave more comments here to demonstrate their relation and how stuff works very clearly to make it easy to pick up for even a tyro.